torchdynamo/variables/tensor.py

```diff
@@ -671,6 +676,17 @@ def from_tensor_variable(cls, tensor_variable, raw_value, need_unwrap=True):
             need_unwrap=need_unwrap,
         )
 
+    def convert_to_constant(self, tx):
+        for graph_arg in tx.output.graphargs:
```

Comment:
this feels rather heavy - matching on source like this for convert_to_constant - for example, when you do slice, you end up doing k (where k is size of slice) * g (where g is number of graphargs) operations. 

You could instead just make a source->grapharg mapping, right? and then do something like `tx.output.erase(self.source)` which would do `src_to_grapharg[self.source].erase() under the hood` and remove the key?

Reply:
I think slice doesn't work as you described, because:
* If we support slice as unspec, 3 graphargs(start, stop, step) at most.
* Actually we don't support slice as unspec.

To have a source->grapharg mapping makes sense, this is aligned with my next step, see my [comments here](https://github.com/pytorch/torchdynamo/pull/714#discussion_r938293298).

Reply:
```
 [
                    x.convert_to_constant(self)
                    if isinstance(x, UnspecializedPythonVariable)
                    else x
                    for x in items
                ],
```

Looks like we get a potential to call `convert_to_constant` on every item, which in turn does this heavy iteraiton. Would you mind putting the mapping in this PR, or even as a PR before this one?

Reply:
```
class SliceVariable(BaseListVariable):
  def __init__(self, items, **kwargs):
          start, stop, step = [variables.ConstantVariable(None)] * 3
          if len(items) == 1:
              (stop,) = items
          elif len(items) == 2:
              start, stop = items
          elif len(items) == 3:
              start, stop, step = items
          else:
              assert False
          super().__init__([start, stop, step], **kwargs)
```
Look at ```SliceVariable``` constructor function, ```len(items)``` should always <= 3.

Reply:
Still seems wasteful when it doesn't really need to for loop, but I wont fight you on it.